File: lib/java/src/main/java/io/github/msgpact/internal/types/VString.java
package io.github.msgpact.internal.types;

import static io.github.msgpact.internal.generation.GenerateRandomString.generateRandomString;
import static io.github.msgpact.internal.validation.ValidateString.validateString;

import java.util.List;

import io.github.msgpact.internal.generation.GenerateContext;
import io.github.msgpact.internal.validation.ValidateContext;
import io.github.msgpact.internal.validation.ValidationFailure;

public class VString implements VType {
    public static final String _STRING_NAME = "String";

    @Override
    public int getTypeParameterCount() {
        return 0;
    }

    @Override
    public List<ValidationFailure> validate(Object value, List<VTypeDeclaration> typeParameters, ValidateContext ctx) {
        return validateString(value);
    }

    @Override
    public Object generateRandomValue(Object blueprintValue, boolean useBlueprintValue,
            List<VTypeDeclaration> typeParameters, GenerateContext ctx) {
        return generateRandomString(blueprintValue, useBlueprintValue, ctx);
    }

    @Override
    public String getName() {
        return _STRING_NAME;
    }
}

File: lib/ts/src/internal/types/VString.ts
import { VTypeDeclaration } from './VTypeDeclaration';
import { ValidationFailure } from '../validation/ValidationFailure';
import { VType } from './VType';
import { validateString } from '../validation/ValidateString';
import { generateRandomString } from '../generation/GenerateRandomString';
import { GenerateContext } from '../generation/GenerateContext';
import { ValidateContext } from '../validation/ValidateContext';

export const stringName: string = 'String';

export class VString extends VType {
    getTypeParameterCount(): number {
        return 0;
    }

    validate(value: any, typeParameters: VTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return validateString(value);
    }

    generateRandomValue(
        blueprintValue: any,
        useBlueprintValue: boolean,
        typeParameters: VTypeDeclaration[],
        ctx: GenerateContext,
    ): any {
        return generateRandomString(blueprintValue, useBlueprintValue, ctx);
    }

    getName(): string {
        return stringName;
    }
}
